docs/Experiments.rst

```diff
@@ -0,0 +1,243 @@
+| **Data**   | **Task**                | **Link**   | **#Data**       | **#Feature**   |
++============+=========================+============+=================+================+
+| Criteo     | Binary classification   | `link`_    | 1,700,000,000   | 67             |
++------------+-------------------------+------------+-----------------+----------------+
+
+This data contains 13 integer features and 26 category features of 24 days click log.
+We statistic the CTR and count for these 26 category features from the first ten days,
+then use next ten days' data, which had been replaced the category features by the corresponding CTR and count, as training data.
+The processed training data hava total 1.7 billions records and 67 features.
+
+Environment
+^^^^^^^^^^^
+
+We use 16 Windows servers as experiment platform, details are listed in following table:
+
++----------------------+-----------------+----------------------+-------------------------------+
+| **OS**               | **CPU**         | **Memory**           | **Network Adapter**           |
++======================+=================+======================+===============================+
+| Windows Server 2012  | 2 * E5-2670 v2  | DDR3 1600Mhz, 256GB  | Mellanox ConnectX-3, 54Gbps,  |
+|                      |                 |                      | RDMA support                  |
++----------------------+-----------------+----------------------+-------------------------------+
+
+Settings
+^^^^^^^^
+
+  .. code::
+
+      learning_rate = 0.1
+      num_leaves = 255
+      num_trees = 100
+      num_thread = 16
+      tree_learner = data
+
+We use data parallel here, since this data is large in ``#data`` but small in ``#feature``.
+
+Other parameters are default values.
+
+Result
+^^^^^^
+
++----------------+---------------------+---------------------------------+
+| **#Machine**   | **Time per Tree**   | **Memory Usage(per Machine)**   |
++================+=====================+=================================+
+| 1              | 627.8 s             | 176GB                           |
++----------------+---------------------+---------------------------------+
+| 2              | 311 s               | 87GB                            |
++----------------+---------------------+---------------------------------+
+| 4              | 156 s               | 43GB                            |
++----------------+---------------------+---------------------------------+
+| 8              | 80 s                | 22GB                            |
++----------------+---------------------+---------------------------------+
+| 16             | 42 s                | 11GB                            |
++----------------+---------------------+---------------------------------+
+
+From the results, we find that LightGBM performs linear speed up in parallel learning.
+
+GPU Experiments
+---------------
+
+Refer to `GPU Performance <./GPU-Performance.rst>`__.
```

Comment:
I've found out that many urls in new doc files (no matter `md` or `rst`) couldn't be resolved: they converted to something like `LightGBM/docs/_build/html/GPU-Performance.rst` or even `https://github.com/Microsoft/LightGBM/tree/master/docsInstallation-Guide.md`.

Maybe someone known WAIDW?

Reply:
The problem starts to be more precise. Now links from only `rst` files are not converted to `html`.

It was just a typo in `md` file which was resulted in broken url.

Reply:
I've understood that `rst` documents are not processed by *recommonmark* and passed as is. So, my solution is based on js function which allows keep actual links at GitHub (e.g. `./Features.md`) and at Read the Docs replace them with `.html`.

Reply:
do you really need the extension of `.rst`?

Reply:
@henry0312 Yeah! I decided to convert from `md` to `rst` after I found your [comment](https://github.com/Microsoft/LightGBM/pull/776#issuecomment-319850209).
Now I have plans (not in this PR) convert all files to `rst`, so we'll be able to create the documentation without **recommonmark** which are not compatible with the latest **Sphinx** (see #958).

Reply:
Ah, I mean I guess

right:

```
`GPU Performance <./GPU-Performance>`__. 
```

wrong:

```
`GPU Performance <./GPU-Performance.rst>`__.
```

(not including `.rst`)

Reply:
Oh, I see.
Unfortunately, such links don't work neither in generated `.html` files, nor at GitHub repo.

Reply:
Oh, really?
but actually https://raw.githubusercontent.com/Microsoft/LightGBM/master/docs/index.rst works fine.
And I don't think we need to support links on GitHub like ReadTheDocs.

Reply:
Maybe, `GPU Performance <GPU-Performance>` is works.

Reply:
No... ;(
![image](https://user-images.githubusercontent.com/25141164/31062945-470b3ea2-a737-11e7-9046-0c542905158c.png)

I spent two whole days to make links work properly, believe me, I tried all variants 😄 .
Why you don't appreciate js function? I think it's rather elegant solution. With it we have proper links at GitHub and at ReadTheDocs.

Reply:
I'm sorry I got the issue.
So, will you cnvert all `md` to `rst` really? 🤣

Reply:
I have the wish to do it with the aim to remove **recommonmark** from the converting chain (shorter chain -> less bugs 😃 (e.g. https://github.com/sphinx-doc/sphinx/issues/3800)). I think, it's rather good motivation.

Reply:
> Why you don't appreciate js function?

because Github repo isn't the place where we provides documents for users, and we don't need to waste time (e.g. maintaining js) to render documents on Github.

Reply:
> I have the wish to do it with the aim to remove recommonmark from the converting chain (shorter chain -> less bugs 

I think so 👍

Reply:
>because Github repo isn't the place where we provides documents for users, and we don't need to waste time (e.g. maintaining js) to render documents on Github.

js is for ReadTheDocs, pretty looking GitHub is a free bonus :-)

Reply:
Did you read http://www.sphinx-doc.org/en/stable/markup/inline.html#cross-referencing-documents?
When we don't need any markdowns, we wil be able to put links across documents (.rst) with it, so I think we don't use js.

Reply:
I think 2 lines of js code is not so critical and cannot be called "maintaining js", if they completely solve the problem without any conditions.

Reply:
yeah, the tiny js is really easy to maintain.
If we want to provide docs on GitHub too , I think we shuold use GitHub Pages where we push built html, and use GitHub to maintain the sources of documents.
What do you think about GitHub and GitHub Pages?

Reply:
To be honest, I think that ReadTheDocs + GitHub repo is quite enough. I suppose, GitHub Pages is excess thing, GitHub renders current `rst` files rather good (markdown is saved and all cross links are working between files), you may take a look to my [branch](https://github.com/StrikerRUS/LightGBM/tree/readthedocs/docs) to see it works.

Yeah, I saw this Sphinx documentation. Unfortunately, such style of links doesn't work at GitHub repo - we need to provide file extensions anyway.

Reply:
Great! Please also when you will review my branch
>Please review Installation-Guide.rst with more attention - I added some sections about GUI compilation with Visual Studio there.

Reply:
Thanks for your suggestion! Unfortunately, I had problems with Pandoc installation and decided to use [online variant](https://pandoc.org/try/) with further [validation](http://rst.aaroniles.net/).

Reply:
> Please review Installation-Guide.rst with more attention - I added some sections about GUI compilation with Visual Studio there.

I don't know if the sections are right or wrong because I don't have Windows.
@guolinke, can you help us?